### PR TITLE
Add implementation of FSWatcher and FSScanner for filestream

### PR DESCRIPTION
**Pull request by kvch**
## What does this PR do?

This PR adds the implementation for `FSWatcher` and `FSScanner` for the `filestream` input.

The implementation of `FSScanner` is called `fileScanner`. It is responsible for
* resolves recursive globs on creation
* normalizes glob patterns on creation
* finds files which match the configured paths and returns `FileInfo` for those

This is the refactored version of the `log` input's scanner, globber functions.

The implementation of `FSWatcher` is called `fileWatcher`. It checks the file list returned by `fileScanner` and creates events based on the result.

## Why is it important?

It is required for the `filestream` input.

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
~~- [ ] I have made corresponding changes to the documentation~~
~~- [ ] I have made corresponding change to the default configuration files~~
- [x] I have added tests that prove my fix is effective or that my feature works
~~- [ ] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.~~

## Related issues

Related #20243

### CODE REVIEW

**Comment by kvch** [comment]
jenkins run tests

**Comment by kvch** [comment]
Failing tests are unrelated.

File: filebeat/input/filestream/fswatch.go [hunk 1]
```diff
@@ -0,0 +1,387 @@
...
+
+	return false
+}
+
+func (s *fileScanner) isOriginalAndSymlinkConfigured(file string, paths map[string]os.FileInfo) bool {
+	if s.symlinks {
+		fileInfo, err := os.Stat(file)
+		if err != nil {
+			s.log.Debugf("stat(%s) failed: %s", file, err)
+			return true
```
Comment (faec) [inline filebeat/input/filestream/fswatch.go:362]: `return true` is counterintuitive to me here, as i would have expected it to be false in this case. If `true` is correct, could you add a comment explaining why?

**Review by faec — APPROVED** [review]
Several suggestions, but nothing complex enough to require another review pass, approved.

File: filebeat/input/filestream/fswatch.go [hunk 2]
```diff
@@ -0,0 +1,387 @@
...
+	paths := map[string]os.FileInfo{}
+
+	for _, path := range s.paths {
+		matches, err := filepath.Glob(path)
+		if err != nil {
+			s.log.Errorf("glob(%s) failed: %v", path, err)
+			continue
+		}
+
+	OUTER:
```
[the remark this answers is not in the document]
Comment (kvch) [reply filebeat/input/filestream/fswatch.go:307]: The goto label is just a leftover from the refactored code. In the original, it was required as it had one more for loop inside the second loop. As I have eliminated the innermost loop, it is not necessary anymore. Thank you for catching it!

**Comment by v1v** [comment]
Test failures might not be relevant, but the step failures are relevant, symlinks are breaking the windows builds

**Pull request merged**